plugins/ComposeAuth/src/commonMain/kotlin/io/github/jan/supabase/compose/auth/LoginConfig.kt

```diff
@@ -0,0 +1,71 @@
+package io.github.jan.supabase.compose.auth
+
+import kotlinx.serialization.json.JsonObject
+
+
+/**
+ * Config for [ComposeAuth]
+ */
+interface LoginConfig {
+    /**
+     * Returns clientId for native login
+     */
+    val serverClientId: String
+}
+
+
+/**
+ * Config for requesting IDToken from play-auth API
+ * @param isSupported Sets whether Google ID token-backed credentials should be returned by the API.
+ * @param filterByAuthorizedAccounts Sets whether to only allow the user to select from Google accounts that are already authorized to sign in to your application.
+ * @param associateLinkedAccounts Sets whether to support sign-in using Google accounts that are linked to your users' accounts.
+ * @param nonce Sets the nonce to use when generating a Google ID token.
+ * @param extraData Add extra data for user on sign-in
+ */
+data class GoogleLoginConfig(
+    override val serverClientId: String,
+    val isSupported: Boolean = true,
+    val filterByAuthorizedAccounts: Boolean = false,
+    val associateLinkedAccounts: Pair<String, List<String>>? = null,
+    val nonce: String? = null,
+    var extraData: JsonObject? = null
+) : LoginConfig
+
+/**
+ * Config for Apple's Authorization API
+ * @param nonce A string value to pass to the identity provider.
+ * @param extraData Add extra data for user on sign-in
+ */
+data class AppleLoginConfig(
+    override val serverClientId: String = "",
```

Comment:
```suggestion
    override val serverClientId: String,
```
I don't think that you want the server client id empty